src/modm/driver/display/sh1106_i2c.hpp

```diff
@@ -39,13 +39,10 @@ class Sh1106 : public Ssd1306<I2cMaster, Height>
 
 		this->transaction_success = true;
 
-		this->commandBuffer[0] = ssd1306::AdressingCommands::HigherColumnStartAddress;
-		this->commandBuffer[1] = 0x02;
-
 		for (page = 0; page < Height / 8; page++)
 		{
-			this->commandBuffer[2] = 0xB0 | page;
-			this->transaction_success &= RF_CALL(this->writeCommands(3));
```

Comment:
The command sequence {`0x10`, `0x02`, `0xB0 | page`} gets interpreted (at least on my SSD1306 _GME12864_ module) as three command:
1.`0x10`:  _Set Higher Column Start Address for Page Addressing Mode (10h~1Fh)_ **to value 0**
2.`0x02` _Set Lower Column Start Address for Page Addressing Mode (00h~0Fh)_ **to value 2**
3. `0xB0` _Set Page Start Address for Page Addressing Mode (B0h~B7h)_ **to value `page`**

The first two commands result in the first two columns of display pixels displaying garbage and the last two columns of the frame not getting displayed, so I removed them.
SSD1106 should behave the same way like SH1106 in _Page Addressing Mode_, but I don't have SH1106 hardware laying arround, so I can't test it. Maybe @TomSaw could check?

Reply:
And the hardware test shows:
These lines have a purpose. Whitout, the displays pages (columns of pixels) get ugly shifted.
I remember having read about this necessary micro-management in an adafruit or arduino library comment, not in the datasheet or elsewhere.